models/tasks.js

```diff
@@ -193,29 +194,45 @@ const fetchUserTasks = async (username, statuses = [], field, order) => {
   }
 };
 
-const getNewTask = async (skill, level) => {
-  const task = await tasksModel
-    .where("category", "==", skill)
-    .where("level", ">=", level)
-    .where("level", "<=", level + 2)
-    .where("status", "==", TASK_STATUS.AVAILABLE)
-    .limit(1)
-    .get();
+const getNewTask = async (skill = undefined, level = undefined) => {
+  const availableTasks = await tasksModel.where("status", "==", TASK_STATUS.AVAILABLE).get();
+  const idArray = [];
 
-  let taskData, id;
-  if (!task.empty) {
-    task.forEach((doc) => {
-      id = doc.id;
-      taskData = doc.data();
-    });
-    return {
-      task: {
-        id,
-        ...taskData,
-      },
-    };
+  let task;
+
+  if (!availableTasks.empty) {
+    availableTasks.forEach((item) => idArray.push(item.id));
+
+    if (!skill) {
+      task = await ItemModel.where("itemType", "==", "TASK").where("levelNumber", "<=", 2).get();
+    } else {
+      task = await ItemModel.where("tagName", "==", skill)
+        .where("itemType", "==", "TASK")
+        .where("levelNumber", ">=", level)
+        .where("levelNumber", "<=", level + 2)
+        .get();
+    }
```

Comment:
Are we not limiting the data here?

Reply:
no, as discussed I will be doing enhancement in other PR

Reply:
Can you put the issue link here please 😄

Reply:
https://github.com/Real-Dev-Squad/website-backend/issues/843